src/harness/harness.ts

```diff
@@ -1725,13 +1741,13 @@ namespace Harness {
             return result;
         }
 
-        export function *iterateOutputs(outputFiles: Harness.Compiler.GeneratedFile[]): IterableIterator<[string, string]> {
+        export function *iterateOutputs(outputFiles: ReadonlyArray<documents.TextDocument>): IterableIterator<[string, string]> {
             // Collect, test, and sort the fileNames
-            outputFiles.sort((a, b) => ts.compareStringsCaseSensitive(cleanName(a.fileName), cleanName(b.fileName)));
+            outputFiles.slice().sort((a, b) => ts.compareStringsCaseSensitive(cleanName(a.file), cleanName(b.file)));
             const dupeCase = ts.createMap<number>();
             // Yield them
             for (const outputFile of outputFiles) {
-                yield [checkDuplicatedFileName(outputFile.fileName, dupeCase), "/*====== " + outputFile.fileName + " ======*/\r\n" + outputFile.code];
+                yield [checkDuplicatedFileName(outputFile.file, dupeCase), "/*====== " + outputFile.file + " ======*/\r\n" + core.removeByteOrderMark(outputFile.text)];

```

Comment:
Are we explicitly not baselining a BOM anymore (even in tests which are only there to check if we output one correctly)?

Reply:
We never added the BOM previously but instead had a `writeByteOrderMark` property on the output object. The new code adds the byte order mark (if needed) to the output file, but strips it here to align with existing output.

Reply:
Huh. OK.

Reply:
To further clarify, previously `outputFile` had this interface:

```ts
        export interface GeneratedFile {
            fileName: string;
            code: string;
            writeByteOrderMark: boolean;
        }
```
The `code` property was the output without the BOM, while `writeByteOrderMark` stored whether a BOM should have been written. The new data structure I'm using stores the full output text that would have been written to the file system including the BOM.